overlay.d/05rhcos/usr/libexec/rhcos-fix-root

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+set -euxo pipefail
+
+if [ -f /run/device_mod ]; then
+    exit
+fi
+
+# check for the existence of the dm-linear mapping
+if [ -h "/dev/mapper/coreos-luks-root-nocrypt" ]; then
+    dm_device=$(readlink /dev/mapper/coreos-luks-root-nocrypt -f)
+    dm_device_name=$(basename $dm_device)
+    partition=$(ls /sys/block/$dm_device_name/slaves/)
+    parent=$(basename "$(lsblk --noheadings --output PKNAME --paths /dev/$partition | head -n1)")
+    partnum=$(cat /sys/block/$parent/$partition/partition)
+    # hard-code the label to root
+    label="root"
+    guid=$(lsblk "/dev/$partition" -no PARTUUID)
+    # move the start sector 32768 to move past the LUKS header
+    startsector="$(($(cat /sys/block/$parent/$partition/start)+32768))"
+    # the dm device is already the correct size
+    size=$(cat /sys/block/$dm_device_name/size)
+
+    # check / handle 4k sector size devices
+    sector_size=$(cat /sys/block/$parent/queue/logical_block_size)
+    if [ "$sector_size" == "4096" ]; then
+        startsector=$((startsector / 8))
+        size=$((size / 8))
+    fi
+
+    # validate that the new partition will start at the xfs super block
+    skipcount=$(($startsector * ($sector_size/4)))
+    [ $(dd if=/dev/$parent bs=4 count=1 status=none skip=$skipcount) == XFSB ]
```

Comment:
Let's add a proper `if` here, because if this test fails, it'll exit nonzero without a log message.

`count=4 bs=1` would allow avoiding the awkward division by 4 on the previous line.